Added blog on contributor workflow

tvvignesh opened the pull request:
This PR adds the blog on contributor workflow as suggested by @Urigo

Urigo commented on pages/blog/oss-contributor-workflow.mdx:
```diff
@@ -0,0 +1,88 @@
+
+While all this might seem very simple, doing this has a long term benefit of communicating your expectations as a maintainer to the community when they are looking to contribute. Now, this is something which some of the projects tend to ignore.
+
+Then we started thinking about the next important thing which we can address to make both maintainer's and contributor's lives easier - LABELS
```
Maybe talk about labels before issue templates?
It's a bigger change

Urigo replied:
with header that's called something like "Process based Github Labels" or something?

Urigo commented on pages/blog/oss-contributor-workflow.mdx:
"The goal here was to make it more obvious for anyone who arrives to an issue where exactly it stands and most importantly, what can you do to help advance the solution"

Urigo commented on pages/blog/oss-contributor-workflow.mdx:
link to the workflow

Urigo commented on pages/blog/oss-contributor-workflow.mdx:
most/almost?

tvvignesh replied:
Thanks for catching that. It should be either `almost all` or `most`

Urigo reviewed (changes requested):
Great post!

I've added some small notes.

I think the most important thing for me to emphasize is that what we really care about and trying to solve is making it easier for people who never contributed to open source to start participating, opening that world to them.

The initial step, the first time is the hardest.

We want to show them the path, what's possible and how to get there.

If you can add a bit more about these types of things it would be awesome!

kamilkisiela commented:
Preview: https://the-guild-webs-git-fork-tvvignesh-blog-contrib-workflow-15224a.vercel.app/blog/oss-contributor-workflow

tvvignesh commented on pages/blog/oss-contributor-workflow.mdx:
```diff
@@ -0,0 +1,88 @@
+
+Open source has revolutionized the way we build, collaborate and release software and there are definitely a lot of instances where the community of contributors and maintainers have taken the lead to make the projects thrive in the ecosystem and this is really exciting to see.
+
+But as anything goes, it is not without its own set of challenges which advocates us to think of interesting solutions. This stands true in our case as well.
```
Not sure why we are talking about the "career" aspect here - is it really relevant?

Urigo replied:
yes maybe not that. I guess my point is around that there are benefits for people to contribute to open source and if its hard it's a shame.... I'm sure you can write something much better than I did :)

tvvignesh commented:
@Urigo Addressed all your feedback except [this](https://github.com/the-guild-org/the-guild-website/pull/346#discussion_r616817413) which I was not sure of.

Urigo commented on pages/blog/oss-contributor-workflow.mdx:
```diff
@@ -0,0 +1,105 @@
+- A [new system](https://github.com/the-guild-org/shared-resources/) for managing our repositories (Process based labeling system, issue templates)
+
+Open source has revolutionized the way we build, collaborate and release software and there are definitely a lot of instances where the community of contributors and maintainers have taken the lead to make the projects thrive in the ecosystem and this is really exciting to see.
+Making open source more approachable has beneficial for the maintainers, users and the contributors themselves - opening new paths for them that seemed harder before.
```
is beneficial?  has benefits?
It's worth trying opening new paths?
something like that?

Urigo commented on pages/blog/oss-contributor-workflow.mdx:
```diff
@@ -0,0 +1,105 @@
+But as anything goes, it is not without its own set of challenges. The number of open source contributors is a tiny fracture compared to the number of users of our libraries.
+
+This advocates us to think of interesting solutions, as we believe many developers are missing out on these opportunities to advance themselves and their careers.
+
```
"We see that improving that is [The Guild's core mission!](https://the-guild.dev/about-us)"

Urigo reviewed (approved):
as usual, you take my wording and make it 10x better :)

Added small notes but at this point it is ready.

We'll publish it tomorrow
